lib/zinnia.js

```diff
@@ -133,11 +188,24 @@ export async function run ({
       console.error(`Zinnia closed all stdio with code ${code ?? '<no code>'}`)
       childProcess.stderr.removeAllListeners()
       childProcess.stdout.removeAllListeners()
+      if (shouldRestart) return
       const err = new Error(`Zinnia exited ${exitReason ?? 'for unknown reason'}`)
       maybeReportCrashToSentry(err)
       throw err
     })()
   ])
+
+  if (shouldRestart) {
+    return run({
```

Comment:
Do you have any concerns about recursively calling `run`? (We don't want to trigger `RangeError: Maximum call stack size exceeded`.)

```
> run = () => { run() }
[Function: run]
> run()
Uncaught RangeError: Maximum call stack size exceeded
    at run (REPL1:1:15)
```

I think we are safe here because `run` is called from an async function, where the stack is unwinded whenever we await a promise.

```
> run = async () => { await Promise.resolve(); return run() }
[AsyncFunction: run]
> await run()
(runs forever)
```

Is it worth adding a code comment to make this explicit to future readers?

Reply:
You're right that this could happen with synchronous functions. Even then it's unlikely we'll update a process _this_ often, but better safe than sorry. I'm adding a code comment